survey-sdk/src/main/java/co/nimblehq/survey/sdk/network/MoshiBuilderProvider.kt

```diff
@@ -3,28 +3,42 @@ package co.nimblehq.survey.sdk.network
 import co.nimblehq.survey.sdk.entity.AnswerEntity
 import co.nimblehq.survey.sdk.entity.QuestionEntity
 import co.nimblehq.survey.sdk.entity.SurveyEntity
-import com.squareup.moshi.JsonAdapter
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import moe.banana.jsonapi2.JsonApiConverterFactory
+import moe.banana.jsonapi2.Resource
 import moe.banana.jsonapi2.ResourceAdapterFactory
 import retrofit2.converter.moshi.MoshiConverterFactory
 
 object MoshiBuilderProvider {
 
-    fun provideJsonApiFactory(): JsonAdapter.Factory = ResourceAdapterFactory.builder()
-        .add(SurveyEntity::class.java)
-        .add(QuestionEntity::class.java)
-        .add(AnswerEntity::class.java)
+    private val listJsonApiClass = mutableListOf<Class<out Resource>>(
+        SurveyEntity::class.java,
+        QuestionEntity::class.java,
+        AnswerEntity::class.java
+    )
+
+    fun provideJsonApiFactory(): ResourceAdapterFactory = ResourceAdapterFactory.builder()
+        .add(*listJsonApiClass.toTypedArray())
         .build()
 
-    fun provideJsonApiMoshi(moshiBuilder: Moshi.Builder, factory: JsonAdapter.Factory): Moshi =
-        moshiBuilder.add((factory)).build()
+    fun provideMoshiJsonApi(
+        jsonApiFactory: ResourceAdapterFactory,
+        moshiBuilder: Moshi.Builder
+    ): Moshi = moshiBuilder
+        .add((jsonApiFactory)).build()
 
-    fun getJsonApiConverterFactory(moshi: Moshi): JsonApiConverterFactory =
-        JsonApiConverterFactory.create(moshi)
+    fun getJsonApiConverterFactory(moshiJsonApi: Moshi): JsonApiConverterFactory =
+        JsonApiConverterFactory.create(moshiJsonApi)
 
-    fun getConverterFactory(moshiBuilder: Moshi.Builder): MoshiConverterFactory =
-        MoshiConverterFactory.create(moshiBuilder.add(KotlinJsonAdapterFactory()).build())
+    fun addJsonApiClasses(vararg classes: Class<out Resource>) {
```

Comment:
🤔  If I understand right, we only need this method temporarily, don't we? Because the logic really weird here, the sample-app passing `TokenEntity` to the sdk and the `AuthService` from sdk is using `TokenEntity` from the `same-app`.

Reply:
@hoangnguyen92dn No, the `AuthService` is from `sample-app` along with `TokenEntity` so far. This function support `json-api` entity outside the Sdk.
Let say, suppose our Sdk only support `survey-entity`, the `sample-app` want to use its own `json-api` Entity and `service` without redefine the `Retrofit`. It can use the Sdk for this purpose

Reply:
@doannimble 

> the sample-app want to use its own json-api Entity and service without redefine the Retrofit

Hmm. Can you please explain what is the `service` here? 🤔 If I understand right, we are supposing to support handling other API requests/response? 🤔

Reply:
@hoangnguyen92dn  `service` in this example is `AuthService`. We support for both, API request/response and the outside `api service` to help `sample-app` does not need to rebuild `Retrofit` again.

Reply:
@doannimble Imo, we are not building the Networking Sdk or Service Sdk, therefore we should not expose the configuration and handling networking stuff. The SDK is supposed to provide `data` for the developer, in our case is the SurveyList and Survey details with errors for handling outside SDK scope. If it's using for the current sample-app, I think it's acceptable as a temporary solution because it would take us much effort to handle networking on the application scope.

Reply:
@hoangnguyen92dn Currently, we do not expose the configuration and network handling stuff. Just given more advance in the Sdk. I think we will make a decision to keep or remove it later.

Reply:
@doannimble 👌 If so, please add a comment to mention the purpose and we will finalize the decision later on 🙏

Reply:
@hoangnguyen92dn Agree